vortexor/src/cli.rs

```diff
@@ -0,0 +1,159 @@
+    pub max_fwd_unstaked_connections: String,
+    pub max_streams_per_ms: String,
+    pub max_connections_per_ipaddr_per_min: String,
+    pub num_quic_endpoints: String,
+}
+
+impl Default for DefaultArgs {
+    fn default() -> Self {
+        Self {
+            bind_address: "0.0.0.0".to_string(),
+            dynamic_port_range: format!("{}-{}", VALIDATOR_PORT_RANGE.0, VALIDATOR_PORT_RANGE.1),
+            max_connections_per_peer: MAX_QUIC_CONNECTIONS_PER_PEER.to_string(),
+            max_tpu_staked_connections: MAX_STAKED_CONNECTIONS.to_string(),
+            max_tpu_unstaked_connections: MAX_UNSTAKED_CONNECTIONS.to_string(),
+            max_fwd_staked_connections: MAX_STAKED_CONNECTIONS
+                .saturating_add(MAX_UNSTAKED_CONNECTIONS)
+                .to_string(),
+            max_fwd_unstaked_connections: 0.to_string(),
+            max_streams_per_ms: DEFAULT_MAX_STREAMS_PER_MS.to_string(),
+            max_connections_per_ipaddr_per_min: DEFAULT_MAX_CONNECTIONS_PER_IPADDR_PER_MINUTE
+                .to_string(),
+            num_quic_endpoints: NUM_QUIC_ENDPOINTS.to_string(),
+        }
+    }
+}
+
+fn port_range_validator(port_range: String) -> Result<(), String> {
+    if let Some((start, end)) = solana_net_utils::parse_port_range(&port_range) {
+        if end.saturating_sub(start) < MINIMUM_VALIDATOR_PORT_RANGE_WIDTH {
+            Err(format!(
+                "Port range is too small.  Try --dynamic-port-range {}-{}",
+                start,
+                start.saturating_add(MINIMUM_VALIDATOR_PORT_RANGE_WIDTH)
+            ))
+        } else if end.checked_add(QUIC_PORT_OFFSET).is_none() {
+            Err("Invalid dynamic_port_range.".to_string())
+        } else {
+            Ok(())
+        }
+    } else {
+        Err("Invalid port range".to_string())
+    }
+}
+
+pub fn app<'a>(version: &'a str, default_args: &'a DefaultArgs) -> App<'a, 'a> {
+    return App::new(crate_name!())
+        .about(crate_description!())
+        .version(version)
+        .global_setting(AppSettings::ColoredHelp)
+        .global_setting(AppSettings::InferSubcommands)
+        .global_setting(AppSettings::UnifiedHelpMessage)
+        .global_setting(AppSettings::VersionlessSubcommands)
+        .arg(
+            Arg::with_name("identity")
+                .short("i")
+                .long("identity")
+                .value_name("KEYPAIR")
+                .takes_value(true)
+                .validator(is_keypair_or_ask_keyword)
+                .help("Vortexor identity keypair"),
+        )
+        .arg(
+            Arg::with_name("bind_address")
+                .long("bind-address")
+                .value_name("HOST")
+                .takes_value(true)
+                .validator(solana_net_utils::is_host)
+                .default_value(&default_args.bind_address)
+                .help("IP address to bind the validator ports"),
+        )
+        .arg(
+            Arg::with_name("dynamic_port_range")
```

Comment:
why does it need dynamic port range?

Reply:
This is similar to how it is done in the validator side which allows the user to specify the range for the tpu addresses.